### ISSUE TITLE
the spawn system does not properly account for world borders.

### ISSUE THREAD

**Issue by PATXS** [Multiverse/Multiverse-Core#2638]
### Information

* **Server version:** `This server is running Paper version git-Paper-769 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)` [fresh install, multiverse is the only plugin installed]

* **Full output of `/mv version -p`:** https://paste.gg/p/anonymous/dfd09511b2a540d99fbed93c861ba602 (there was no paste.gg link output, so i uploaded it myself. this is the latest snapshot version as of reporting, but i originally ran into this bug in stable)

* **Server log:** https://gist.github.com/PATXS/5e35d9337036c21aad9a36dacac39410

### Details
i was **able** to reproduce my issue on a freshly setup and up-to-date server with the latest version of Multiverse plugins with no other plugins and with no kinds of other server or client mods.

**Description**  
[for context, all the testing here was done with a border size of 192 in width (`worldborder set 192` without manually setting any center), since it is not an unreasonable size but still extremely likely to require spawn readjustment. vanilla seems to have no problems with this, for reference, but i couldn't test teleporting into a separate world's spawn in vanilla without using another world manager.]

multiverse's spawning system and the "safe spawn" feature seem to be unprepared to handle worlds containing borders. when dealing with a newly added 192-wide border on an already loaded (but never visited) world, multiverse can spawn the player outside of the border the first time, no matter the safe spawn setting. this kills the player on default settings. when a multiverse-assigned "safe spawn" is enabled for that world, it can _continously_ respawn the player outside of the world border and therefore kill them repeatedly, forever. when the safe spawn feature is disabled, the game seems to be able to respawn the player inside the border (but keep in mind this requires the player to die once first).

when importing a never-visited world that already contains a 192-wide border and an unmodified spawnpoint, multiverse seems to always spawn the player below bedrock (and into the void) on first visit. it seems to think the world spawn is 0,0,0 (and it outputs this in console!), and the adjusted spawn does not deviate much from that. once again, if the safe adjusted spawn is enabled, it will keep spawning them here and render the world entirely unplayable unless players are manually teleported up. if adjusted spawn is disabled, the game respawns the player somewhere safe after death. [note: this issue doesn't only happen when importing a world via a command. if the world in question is the server's primary world (just named "world"), multiverse will still do this. when starting the server up without any plugins, this issue is nowhere to be seen. the same can be said when loading the world into singleplayer.]

**Steps to reproduce**  
1. create a world named "test" with multiverse
2. set a border with `execute in minecraft:test run worldborder set 192`
3. teleport yourself there and likely die

OR

1. create a world in another server environment, and set a border with `execute in minecraft:overworld run worldborder set 192`
2. copy the world folder to a server with multiverse
3. import it manually or load it as the main world
4. spawn below bedrock

**Expected behavior**  
the player should be spawned within the world borders at the very least, and not into the void. it's hard to tell whether some of this is vanilla behavior or not, but the vanilla game (as well as paper by itself) does seem to flawlessly handle first-time spawns into the very same worlds i tested with multiverse.

and i apologize if this is not appropriate to mention, but the world manager "hyperverse" handles all of these spawn situations flawlessly as well, and could be a good reference here. i suspect it just lets the game handle spawning as it normally would, but i have no proof of that and don't know how it would be done.

i have no screenshots nor videos but can provide if necessary.

**Comment by benwoo1110** [comment]
May I ask how are you teleporting players?

**Comment by benwoo1110** [comment]
Also the it says `New Spawn for 'importtest' is located at: importtest:1.50,-2.00,0.50:0.00:0.00` Which I'm sure its within 192 radius. Tho yea, block safety code probably doesn't account for world borders. But if you set spawn to the correctly location before implementing world borders it shouldn't be an issue.

**Comment by PATXS** [comment]
lol yes it is within the 192 block radius, but that spawn is below bedrock at a Y level of -2

the reason i found out about this is that for my admittedly niche case, i was automatically generating new worlds and setting up borders before players even got there. so it's not necessarily that i need this fixed, since i can just stop doing it like that, but these two possible scenarios i reported are pretty broken and i thought they might deserve a little bit of dev attention.

**Comment by benwoo1110** [comment]
Yes I see the areas that can be improve, but current not a huge amount of time in hand. PR will be linked if someone can work on it.
